engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportSizeOf.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.stateChanges;
+
+import org.terasology.rendering.dag.FBOManagerSubscriber;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.StateChange;
+import org.terasology.rendering.dag.tasks.SetViewportSizeOfTask;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FrameBuffersManager;
+
+/**
+ * TODO: Add javadocs
+ */
+public final class SetViewportSizeOf implements FBOManagerSubscriber, StateChange {
+    private static final String DEFAULT_FBO_NAME = "sceneOpaque";
+    private static SetViewportSizeOf defaultInstance = new SetViewportSizeOf();
+    private SetViewportSizeOfTask task;
+
+    private FBO fbo;
+    private String fboName;
+    private FrameBuffersManager frameBuffersManager;
+
+    public SetViewportSizeOf(String fboName, FrameBuffersManager frameBuffersManager) {
+        this.fboName = fboName;
+        this.frameBuffersManager = frameBuffersManager;
+        fbo = frameBuffersManager.getFBO(fboName);
+    }
+
+    public SetViewportSizeOf() {
+        this.fboName = DEFAULT_FBO_NAME;
+    }
+
+    @Override
+    public StateChange getDefaultInstance() {
+        // TODO: a very dirty approach :(, however defaultInstance requires a frameBuffer instance, any suggestions?
+        if (defaultInstance.frameBuffersManager == null) {
+            defaultInstance.frameBuffersManager = frameBuffersManager;
+            defaultInstance.fbo = frameBuffersManager.getFBO(DEFAULT_FBO_NAME);
+        }
+        return defaultInstance;
+    }
+
+    @Override
+    public RenderPipelineTask generateTask() {
+        if (task == null) {
+            task = new SetViewportSizeOfTask(fboName, fbo.width(), fbo.height());
+            frameBuffersManager.subscribe(this);
+        } else {
+            update();
```

Comment:
I'm ok with leaving this as it is, for safety, but I do wonder if the update() is really necessary here. I mean: the update heregets executed only if the task is not null. But that means the state change is subscribed to the fbm. And that means that whenever the buffer change the update gets called anyway, but only if needed.